spring-integration-core/src/main/java/org/springframework/integration/config/IntegrationRegistrar.java

```diff
@@ -414,14 +423,91 @@ private void registerDefaultDatatypeChannelMessageConverter(BeanDefinitionRegist
 	 * @param registry the registry.
 	 */
 	private void registerArgumentResolverMessageConverter(BeanDefinitionRegistry registry) {
-		if (!registry.containsBeanDefinition(IntegrationContextUtils.ARGUMENT_RESOLVER_MESSAGE_CONVERTER_BEAN_NAME)) {
-			BeanDefinitionBuilder postProcessorBuilder = BeanDefinitionBuilder
+		if (!registry.containsBeanDefinition(
+				IntegrationContextUtils.ARGUMENT_RESOLVER_MESSAGE_CONVERTER_BEAN_NAME)) {
+			BeanDefinitionBuilder resolverBuilder = BeanDefinitionBuilder
 					.genericBeanDefinition(ConfigurableCompositeMessageConverter.class);
 			registry.registerBeanDefinition(IntegrationContextUtils.ARGUMENT_RESOLVER_MESSAGE_CONVERTER_BEAN_NAME,
-					postProcessorBuilder.getBeanDefinition());
+					resolverBuilder.getBeanDefinition());
+		}
+	}
+
+	/**
+	 * Register the default {@link MessageHandlerMethodFactory} for handler method
+	 * invocation.
+	 * @param registry the registry.
+	 */
+	private void registerArgumentResolvers(BeanDefinitionRegistry registry) {
+		if (!registry.containsBeanDefinition(IntegrationContextUtils.ARGUMENT_RESOLVERS_BEAN_NAME)) {
+			registry.registerBeanDefinition(IntegrationContextUtils.ARGUMENT_RESOLVERS_BEAN_NAME,
+					internalArgumentResolversBuilder(registry, false).getBeanDefinition());
 		}
 	}
 
+	/**
+	 * Register the default {@link MessageHandlerMethodFactory} for handler method
+	 * invocation for lists.
+	 * @param registry the registry.
+	 */
+	private void registerListCapableArgumentResolvers(BeanDefinitionRegistry registry) {
+		if (!registry.containsBeanDefinition(
+				IntegrationContextUtils.LIST_ARGUMENT_RESOLVERS_BEAN_NAME)) {
+			registry.registerBeanDefinition(IntegrationContextUtils.LIST_ARGUMENT_RESOLVERS_BEAN_NAME,
+					internalArgumentResolversBuilder(registry, true).getBeanDefinition());
+		}
+	}
+
+	private BeanDefinitionBuilder internalArgumentResolversBuilder(BeanDefinitionRegistry registry,
+			boolean listCapable) {
+		ManagedList<BeanReference> resolvers = new ManagedList<>();
+		if (!registry.containsBeanDefinition(IntegrationContextUtils.PAYLOAD_RESOLVER_BEAN_NAME)) {
```

Comment:
Why do we need to have the all as beans?

I might agree that we have to register two sets of resolvers - single entry and listCapable, but I don't see reason to register each `ArgumentResolver` as a bean.

Seems for me just a simple `BeanDefinition` as reference to the `ManagedList` does the trick and those "ephemeral" beans are populated with the `BeanFactory` if that...

Thanks